cdap-tms/src/main/java/co/cask/cdap/messaging/store/hbase/HBaseTableFactory.java

```diff
@@ -98,10 +99,20 @@
     this.coprocessorManager = new CoprocessorManager(cConf, locationFactory, tableUtil);
 
     RejectedExecutionHandler callerRunsPolicy = new RejectedExecutionHandler() {
+
+      private final AtomicInteger rejected = new AtomicInteger(0);
+      private int logFreq = 1;
+
       @Override
       public void rejectedExecution(Runnable r, ThreadPoolExecutor executor) {
-        LOG.info("No more threads in the HBase scan thread pool. Consider increase {}. Runnable from caller thread {}",
-                 Constants.MessagingSystem.HBASE_MAX_SCAN_THREADS, Thread.currentThread().getName());
+        // Exponentially log less
+        if (rejected.incrementAndGet() % logFreq == 0) {
+          LOG.info("No more threads in the HBase scan thread pool. Consider increase {}. Scan from caller thread {}",
```

Comment:
if this gets logged like ten times, and then we don't hit this issue anymore... for ten days, and then it happens on the tenth day, it wont get logged then. We probably want to see the info log if it happens then?